server/src/test/java/org/elasticsearch/search/aggregations/pipeline/AvgBucketAggregatorTests.java

```diff
@@ -153,51 +154,23 @@ public void testComplicatedBucketPath() throws IOException {
                     document.add(new SortedNumericDocValuesField(DATE_FIELD, asLong(date)));
                     document.add(new SortedNumericDocValuesField(VALUE_FIELD, randomInt()));
                     document.add(new SortedSetDocValuesField(textField, new BytesRef("value")));
-                    document.add(
-                        new KeywordFieldMapper.KeywordField(textField, new BytesRef("value"), KeywordFieldMapper.Defaults.FIELD_TYPE)
-                    );
                     indexWriter.addDocument(document);
                     document.clear();
                 }
             }
 
-            InternalAvg avgResult;
-            InternalDateHistogram histogramResult;
             InternalFilter filterResult;
-            InternalTerms<?, ?> internalTerms;
             try (IndexReader indexReader = DirectoryReader.open(directory)) {
-                IndexSearcher indexSearcher = newSearcher(indexReader, true, true);
+                IndexSearcher indexSearcher = newIndexSearcher(indexReader);
```

Comment:
This is the key change. I was using the newSearcher method from lucene tests which randomly wraps it as a parallel reader, which we don't support.